core/src/mender-deployment-data.c

```diff
@@ -279,6 +282,41 @@ __mender_deployment_data_set_string(mender_deployment_data_t *deployment_data, c
     return MENDER_OK;
 }
 
+mender_err_t
+__mender_deployment_data_get_state(mender_deployment_data_t *deployment_data, const char *key, mender_update_state_t *state) {
+
+    if (NULL == deployment_data) {
+        return MENDER_NOT_FOUND;
+    }
+
+    assert(NULL != key);
+    assert(NULL != state);
+
+    cJSON *item;
+    if (NULL == (item = cJSON_GetObjectItemCaseSensitive(deployment_data, key))) {
+        return MENDER_FAIL;
+    }
+
+    /* Assign state value to state pointer */
+    double state_value = cJSON_GetNumberValue(item);
```

Comment:
What happens if `item` is JSON null?

Reply:
That is checked in the if-statement above